web/src/components/Login.js

```diff
@@ -0,0 +1,90 @@
+import { useState } from 'react';
+
+// style imports
+import '../styles/login.css';
+import '../styles/homepage.css';
+import RegisterUser from './RegisterUser';
+
+const Login = () => {
+  // set up use states for username, password, response message, and validations
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
+  const [valid, setValid] = useState(false);
+
+  // allow user to register instead of logging in
+  const [registrationPopup, setRegistrationPopup] = useState(false);
+
+  // toggle popup as needed
+  const toggleRegistrationPopup = () => {
+    setRegistrationPopup(!registrationPopup);
+  }
+
+  // handle user submitting username and password
+  const handleSubmit = async (e) => {
+    // prevent page from reloading
+    e.preventDefault();
+
+    // try to get response from backend by requesting login verification in json format
+    try {
+      const response = await fetch('/service/login', {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json"
+        },
+        body: JSON.stringify({"username": username, "password": password})
+      });
+
+      // await json response
+      const data = await response.json();
+
+      // check if response ok and then add to localstorage
+      if (response.ok) {
+        localStorage.setItem("status", "true");
+        localStorage.setItem("username", username);
+      } else {
+        setMessage(data.message || "Error logging in.");
+      }
+
+      // set message and validation to confirm or deny login
+      setMessage(data.message);
+      setValid(data.exists);
+
+    } catch (error) {
+      console.log("Error:", error);
+      setMessage("Trouble connecting to backend.");
```

Comment:
Maybe make this "Error logging in" too since the user shouldn't know about how it works.

Reply:
Agreed